### PR TITLE
DBMS/DBMS-Session3 assignments

### PR DESCRIPTION
**Pull request by meta-ramjilal-choudhary**
I did not get logic for 4 queries 
But remaining completed.

@metaamitasharma

### CODE REVIEW

File: DBMS/DBMS-Session3/assignment3_sql.txt [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+ORDER BY total_orders DESC;
+
+
+#Display Monthly sales revenue of the StoreFront for last 6 months. It should display each month’s sale
+#? ? ?
+#-----
+#? ? ?
+
+
+#Mark the products as Inactive which are not ordered in last 90 days
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session3/assignment3_sql.txt]: update product p
left join order_line l on p.product_id NOT IN (SELECT DISTINCT product_id from order_line WHERE order_date < today-90)
set p.isActive = false;

File: DBMS/DBMS-Session3/assignment3_sql.txt [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+SELECT
+	p.prod_id  , (
+				SELECT COUNT(l.order_line_id) 
+                FROM order_line l 
+                WHERE p.prod_id = l.prod_id 
+                AND l.order_status LIKE "Cancelled") AS total_cancel
+FROM 
+	Product p
+ORDER BY total_cancel DESC;
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session3/assignment3_sql.txt]: LIMIT 10

File: DBMS/DBMS-Session3/assignment4_sql.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
+SHOW DATABASES;
+USE ShopCart;
+SHOW TABLES;
+
+CREATE TABLE zip_code_list(
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session3/assignment4_sql.txt]: Create separate tables for state city and code zip for a city is specific for that city.

File: DBMS/DBMS-Session3/assignment2_sql.txt [hunk 4]
```diff
@@ -0,0 +1,72 @@
+SHOW DATABASES;
+USE ShopCart;
+SHOW TABLES;
+
+#Display the list of products (Id, Title, Count of Categories) which fall in more than one Categories
```
[the remark this answers is not in the document]
Comment (meta-ramjilal-choudhary) [reply DBMS/DBMS-Session3/assignment2_sql.txt]: updated

File: DBMS/DBMS-Session3/assignment2_sql.txt [hunk 5]
```diff
@@ -3,63 +3,29 @@ USE ShopCart;
...
-    
-SELECT 
-	prod.prod_id, prod.prod_title, cnt_cat.count_cat
-FROM 
-	Product prod, count_category cnt_cat
-WHERE 
-	prod.prod_id = cnt_cat.prod_id
-    AND
-	cnt_cat.count_cat >1;
+SELECT prod_id
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session3/assignment2_sql.txt]: Query fields Id, Title, Count of Categories mentioned in question.

File: DBMS/DBMS-Session3/assignment2_sql.txt [hunk 6]
```diff
@@ -3,63 +3,29 @@ USE ShopCart;
...
+FROM Product_category 
+GROUP BY prod_id
+HAVING COUNT(prod_id) > 1;
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session3/assignment2_sql.txt]: COUNT(cat_id) I think.

File: DBMS/DBMS-Session3/assignment3_sql.txt [hunk 7]
```diff
@@ -3,78 +3,71 @@ USE ShopCart;
...
-
-SELECT
-	p2.prod_id, 
-    p2.prod_title,
-    (SELECT SUM(l.prod_quantity) 
-    FROM order_line l 
-    WHERE p1.prod_id = l.prod_id) AS total_orders
-FROM
-	products p1, product p2
+SELECT p.prod_id, p.prod_title, COUNT(order_line_id)
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session3/assignment3_sql.txt]: should be sum of product_quantity on lines.

File: DBMS/DBMS-Session3/assignment3_sql.txt [hunk 8]
```diff
@@ -86,15 +79,13 @@ VALUES
...
-FROM 
-	Product p
-ORDER BY total_cancel DESC;
+SELECT p.prod_id, p.prod_title, COUNT(order_status) cacel_total
+FROM order_line l, product p
+WHERE
+    p.prod_id = l.prod_id
+    AND
+    order_status LIKE "Cancel%"
+GROUP BY l.prod_id;
```
Comment (metaamitasharma) [inline DBMS/DBMS-Session3/assignment3_sql.txt]: order by cacel_total DESC LIMIT 10

**Pull request merged**